src/Bicep.Core/Utils/TemplateEvaluator.cs

```diff
@@ -247,22 +243,24 @@ private static Template EvaluateTemplate(JToken? templateJtoken, JToken? paramet
                 var template = TemplateEngine.ParseTemplate(templateJtoken.ToString());
                 var parameters = ConvertParameters(parametersJToken);
 
-                TemplateEngine.ValidateTemplate(template, EmitConstants.NestedDeploymentResourceApiVersion, deploymentScope);
+                var expectedApiVersion = features is not null ? EmitConstants.GetNestedDeploymentResourceApiVersion(features) : EmitConstants.NestedDeploymentResourceApiVersion;
+
+                TemplateEngine.ValidateTemplate(template, expectedApiVersion, deploymentScope);
 
                 TemplateEngine.ProcessTemplateLanguageExpressions(
                     managementGroupName: config.ManagementGroup,
                     subscriptionId: config.SubscriptionId,
                     resourceGroupName: config.ResourceGroup,
                     template: template,
-                    apiVersion: EmitConstants.NestedDeploymentResourceApiVersion,
+                    apiVersion: expectedApiVersion,
                     inputParameters: new(parameters),
                     metadata: metadata,
                     extensionConfigs: [], // TODO: check this with Kyle
```

Comment:
This was added when we merged the latest Deployments code - could you confirm this is correct?

Reply:
Added code to convert extension configs from params jtoken.